### DISCUSSION TITLE
Composite artifacting simulation on NTSC-RS

### DISCUSSION THREAD

**Discussion by leap0x7b** [ntsc-rs/ntsc-rs#208] [General]
Is it possible to simulate composite artifacts from old computers such as the Apple ][ on NTSC-RS? Here's some examples:
![Orange artifact color, generated from white and black pixels, on the TRS-80 Color Computer](https://github.com/user-attachments/assets/b13d41b5-2b74-4cf3-9799-df407a26b7ce)
![Fragment of an Apple II computer screen, showing the actual pixel data composed of vertical stripes (top) and the resulting colors when seen on an NTSC TV (bottom)](https://github.com/user-attachments/assets/a1dfad31-89e7-4ebf-8eea-2bfa5a3f8f4a)
![CGA 640×200 (left: RGB, right: composite monitor)](https://github.com/user-attachments/assets/4aed3038-1428-4bfb-bdc6-0d9a0e7838b5)
![Example of artwork created with the intent of having individual pixel values horizontally averaged over composite video](https://github.com/user-attachments/assets/3781db3b-68de-4a7a-953d-8d8bc6348766)

**Comment by leap0x7b** [comment]
Something like this https://filthypants.blogspot.com/2017/06/new-ntsc-shaders.html

**Comment by leap0x7b** [comment]
Made something exactly like this. This is the closest that I got without making it look too blurry. The chroma error seems to differ between computers so you can't use the same chroma error that you use for Apple II screenshots on a TRS-80 screenshot.

Raw image:
![Raw image](https://github.com/user-attachments/assets/ce0df4b6-e7cc-4438-9019-f36f68149393)
Real composite:
![Real composite](https://github.com/user-attachments/assets/4a15afb0-da33-4624-944c-f58c22378b38)
NTSC-RS:
![NTSC-RS](https://github.com/user-attachments/assets/7124601b-6a74-49ab-b5aa-78dc8fe4eabf)

[The preset itself](https://github.com/user-attachments/files/17396021/apple.ii.json). It looks accurate enough but feel free to improve it if you want to.

**Comment by leap0x7b** [comment]
Another example with Karateka.

Raw image:
![Raw image](https://github.com/user-attachments/assets/3468e7af-0c79-445b-861d-6fede40c75f1)
Real composite:
![Real composite](https://github.com/user-attachments/assets/902e1e14-c8c9-44e1-b60a-4e043d9e9f60)
NTSC-RS:
![NTSC-RS](https://github.com/user-attachments/assets/afda7dad-ec27-4a36-9945-199660cc1af6)

**Comment by xotmatrix** [comment]
Thank you for figuring this out. I would suggest using the Box filter for Chroma demodulation, and reducing the Luma smear. The Notch filter exposes 14 MHz luma signals in a way that would be filtered out by a television.

**Reply by leap0x7b** [reply]
I made this like 2 years ago and forgot about it but I guess thanks for the tip!
